### ISSUE TITLE
Node 15: The property 'options.publicKeyEncoding.format' is invalid. Received 'jwk'

### ISSUE THREAD

**Issue by D4nte** [libp2p/js-libp2p-crypto#221]
Latest release says that only node 15+ is supported: https://github.com/libp2p/js-libp2p-crypto/releases/tag/v0.21.0

Note that the `package.json` was not updated accordingly:

https://github.com/libp2p/js-libp2p-crypto/blob/508c790c52f15d8194f926b0d56c0069caf4ab40/package.json#L65

Node 15 is actually not supported as we can see when running the CI with 15: https://github.com/D4nte/js-libp2p-crypto/runs/4438572378?check_suite_focus=true

```
TypeError [ERR_INVALID_ARG_VALUE]: The property 'options.publicKeyEncoding.format' is invalid. Received 'jwk'
```

Only node 16 works.

I wonder if it's due to this bug only fixed on node 16: https://github.com/nodejs/node/issues/39205

**Comment by D4nte** [comment]
So actually Node 15 has reach end-of-life support. It may make more sense to just say that the library supports node 16 and above.

**Comment by richburdon** [comment]
I still see this problem using Node 16
```
Code/dxos/libp2p-demo  main ✗                                                                                                                                             ⚑ ✚ ◒
▶ node --version
v16.1.0
Code/dxos/libp2p-demo  main ✗                                                                                                                                             ⚑ ✚ ◒
▶ yarn start
yarn run v1.22.5
$ ts-node ./src/main.ts
TypeError [ERR_INVALID_ARG_VALUE]: The property 'options.publicKeyEncoding.format' is invalid. Received 'jwk'
    at new NodeError (node:internal/errors:363:5)
    at parseKeyFormat (node:internal/crypto/keys:268:9)
    at parseKeyFormatAndType (node:internal/crypto/keys:304:18)
    at parseKeyEncoding (node:internal/crypto/keys:332:7)
    at parsePublicKeyEncoding (node:internal/crypto/keys:371:10)
    at parseKeyEncoding (node:internal/crypto/keygen:125:9)
    at createJob (node:internal/crypto/keygen:161:42)
    at generateKeyPair (node:internal/crypto/keygen:75:15)
    at node:internal/util:335:7
    at new Promise (<anonymous>) {
  code: 'ERR_INVALID_ARG_VALUE'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by D4nte** [comment]
> I still see this problem using Node 16
> 
> ```
> Code/dxos/libp2p-demo  main ✗                                                                                                                                             ⚑ ✚ ◒
> ▶ node --version
> v16.1.0
> Code/dxos/libp2p-demo  main ✗                                                                                                                                             ⚑ ✚ ◒
> ▶ yarn start
> yarn run v1.22.5
> $ ts-node ./src/main.ts
> TypeError [ERR_INVALID_ARG_VALUE]: The property 'options.publicKeyEncoding.format' is invalid. Received 'jwk'
>     at new NodeError (node:internal/errors:363:5)
>     at parseKeyFormat (node:internal/crypto/keys:268:9)
>     at parseKeyFormatAndType (node:internal/crypto/keys:304:18)
>     at parseKeyEncoding (node:internal/crypto/keys:332:7)
>     at parsePublicKeyEncoding (node:internal/crypto/keys:371:10)
>     at parseKeyEncoding (node:internal/crypto/keygen:125:9)
>     at createJob (node:internal/crypto/keygen:161:42)
>     at generateKeyPair (node:internal/crypto/keygen:75:15)
>     at node:internal/util:335:7
>     at new Promise (<anonymous>) {
>   code: 'ERR_INVALID_ARG_VALUE'
> }
> error Command failed with exit code 1.
> info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> ```

Update your node install. https://github.com/nodejs/node/issues/39205 was initially reported on 16.4 so fixed after that and you have 16.1.

**Comment by richburdon** [comment]
Wow, thanks so much for the quick response.
